pkg/chains/formats/slsa/internal/material/material.go

```diff
@@ -77,38 +75,28 @@ func PipelineMaterials(ctx context.Context, pro *objects.PipelineRunObject) ([]c
 		}
 		mats = append(mats, m)
 	}
-	pSpec := pro.Status.PipelineSpec
-	if pSpec != nil {
-		pipelineTasks := append(pSpec.Tasks, pSpec.Finally...)
-		for _, t := range pipelineTasks {
-			tr := pro.GetTaskRunFromTask(t.Name)
-			// Ignore Tasks that did not execute during the PipelineRun.
-			if tr == nil || tr.Status.CompletionTime == nil {
-				logger.Infof("taskrun status not found for task %s", t.Name)
-				continue
-			}
 
-			stepMaterials, err := FromStepImages(tr.Status.Steps)
-			if err != nil {
-				return mats, err
-			}
-			mats = append(mats, stepMaterials...)
+	for _, tr := range pro.GetTaskRuns() {
```

Comment:
May be we do that here?

Reply:
```go
if tr == nil || tr.Status.CompletionTime == nil {
        logger.Infof("taskrun status not found for task %s", t.Name)
        continue
}
```

Reply:
or may be move this logic in `pro.GetTaskRuns` and only return the tasks that are `"good"`?

Reply:
and call it `GetCompletedTaskRuns()`?

Reply:
closed as it's resolved in https://github.com/tektoncd/chains/pull/879#discussion_r1271132211